src/reseeding.rs

```diff
@@ -171,12 +151,21 @@ impl<R: RngCore + SeedableRng, Rsdr: RngCore> RngCore for ReseedingRng<R, Rsdr>
     }
 
     fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error> {
-        self.rng.try_fill_bytes(dest)?;
+        let res1 = self.rng.try_fill_bytes(dest);
         self.bytes_until_reseed -= dest.len() as i64;
-        if self.bytes_until_reseed <= 0 {
-            self.try_reseed()?;
+        let res2 = if self.bytes_until_reseed <= 0 {
+            self.try_reseed()
+        } else { Ok(()) };
+        
+        if let Err(e) = res1 {
+            // In the unlikely event the internal PRNG fails, we don't know
+            // whether this is resolvable; reseed immediately and return
+            // original error kind.
```

Comment:
I think this is a nicely found use of the reseeding wrapper. Does it impact benchmarks? Probably not because we benchmark with large slices to fill...

I think this is unnecessary because just about every PRNG (everything that implements `SeedableRng`) never fails. Personally I would leave it up to the user if he wants to handle it.

Reply:
We don't benchmark this method at all currently, but yes, I might revert the changes to this method.

Reply:
How do you think about this now?

Reply:
Frankly I don't care much, especially since your "block rng" idea might significantly alter this. But eventually I guess we should investigate the performance of `thread_rng` more.

I don't care much about changes to this particular method; I can revert if you like but it's probably a small improvement?

Reply:
I don't think the performance is a big deal, but I am not sure adding this functionality is a good idea. Are there situations where reseeding the RNG when `try_fill` fails is not the right action, or maybe even insecure?

I can't think of any, actually I have a trouble imagining when a CSPRNG can fail...
But I can see some use in this: it allows a wrapped RNG to signal it wants to be reseeded.

Reply:
I have some trouble finishing that Block Rng experiment. It is a bit too big for me. Do you want to help / take over, probably after 0.5 is done?

Reply:
Well, I'll leave this for now then.

Can you create an issue regarding the BlockRng experiment? Sounds like it may not be in time for the 0.5 release but I'll try to take a look.